src/Magnum/Math/Bezier.h

```diff
@@ -0,0 +1,118 @@
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+/** @file
+ * @brief Class @ref Magnum::Math::Bezier
+ */
+
+#include <array>
+#include "Vector.h"
+
+namespace Magnum { namespace Math {
+        /**
+        @brief Bezier
+        @tparam order       Order of Bezier curve
+        @tparam dimensions  Dimensions of the control points
+        @tparam T           Underlying data type
+
+        See <a href="https://en.wikipedia.org/wiki/B%C3%A9zier_curve">Bezier Curve</a>.
+        */
+        template<UnsignedInt order, UnsignedInt dimensions, class T> class Bezier {
+
+        public:
+
+            /** @brief Construct Bezier curve with the given array of control points */
+            explicit Bezier(const std::array<Vector<dimensions, T>, order+1> &points) {
+                _points = points;
+            }
+
+            /**
+             * @brief Divides a Bezier curve into two curves of same order having their own control points.
+             *        De Casteljau's algorithm is used.
+             * @param t The interpolation factor
+             *
+             * @return Array of two Bezier curves of the same order
+             */
+            std::array<Bezier<order, dimensions, T >, 2> subdivide(Float t) const {
+                auto i_points = calculateIntermediatePoints(t);
+                std::array<Vector<dimensions, T>, order + 1> left, right;
+                for(UnsignedInt i=0; i<=order; ++i){
+                    left[i] = i_points[0][i];
+                }
+                for(UnsignedInt i = 0, j = order; j>=0; --j, ++i){
+                    right[i] = i_points[i][j];
+                }
+                return {Bezier<order, dimensions, T>(left), Bezier<order, dimensions, T>(right)};
+            }
+
+            /**
+             * @brief Finds the point in the curve for a given interpolation factor
+             *        De Casteljau's algorithm is used.
+             * @param t The interpolation factor
+             */
+            Vector<dimensions, T> lerp (Float t) const {
+                auto i_points= calculateIntermediatePoints(t);
+                return i_points[0][order];
+            };
+
+        private:
+
+            /**
+             * @brief Calculates and returns all intermediate points generated when using De Casteljau's algorithm
+             * @param t The interpolation factor
+             *
+             */
+            std::array<std::array<Vector<dimensions,T>, order + 1>, order + 1> calculateIntermediatePoints(Float t) const {
+                const auto n = order + 1;
+                std::array<std::array<Vector<dimensions,T>, n>, n> i_points;
+                for (UnsignedInt i = 0; i < n; ++i) {
+                    i_points[i][0] = _points[i];
+                }
+                for (UnsignedInt r = 1; r < n; ++r) {
```

Comment:
Also in the [coding style documentation](http://mosra.cz/blog/corrade-doc/corrade-coding-style.html#corrade-coding-style-cpp-style): magnum code uses `for(...)` rather than `for (...)`, same for `if` and `switch`.